File: app/src/main/java/beans/prj/newsapp/data/AppRepository.java
package beans.prj.newsapp.data;

import android.util.Log;
import beans.prj.newsapp.data.model.News;
import beans.prj.newsapp.data.service.APIService;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import javax.inject.Inject;
import java.util.HashMap;

public class AppRepository {

    private APIService apiService;

    @Inject
    public AppRepository(Retrofit retrofit){
        apiService = retrofit.create(APIService.class);
    }

    public void testCall() {
        HashMap<String, String> option = new HashMap<>();
        option.put("country", "kr");

        Call<News> call = apiService.getTopHeadlinesForTest(option);
        call.enqueue(new Callback<News>() {
            @Override
            public void onResponse(Call<News> call, Response<News> response) {

            }

            @Override
            public void onFailure(Call<News> call, Throwable t) {
                Log.i("BEAN", "onFailure"+ t.getMessage());
            }
        });
    }

}

File: app/src/main/java/beans/prj/newsapp/data/AppRepository.kt
package beans.prj.newsapp.data

import android.util.Log
import beans.prj.newsapp.data.model.News
import beans.prj.newsapp.data.service.APIService
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import java.util.*
import javax.inject.Inject

class AppRepository @Inject constructor(retrofit: Retrofit) {
    private val apiService: APIService
    fun testCall() {
        val option = HashMap<String, String>()
        option["country"] = "kr"
        val call = apiService.getTopHeadlinesForTest(option)
        call!!.enqueue(object : Callback<News?> {
            override fun onResponse(call: Call<News?>, response: Response<News?>) {}
            override fun onFailure(call: Call<News?>, t: Throwable) {
                Log.i("BEAN", "onFailure" + t.message)
            }
        })
    }

    init {
        apiService = retrofit.create(APIService::class.java)
    }
}